Bugfix/Re-evaluate split error

lovaiible opened the pull request:
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~~I documented my source code using the JavaDoc / JSDoc style.~~
- [x] ~~I added integration test cases for the server (Spring) related to the features~~
- [x] ~~I added integration test cases for the client (Jest) related to the features~~
- [x] ~~I added screenshots/screencast of my UI changes~~
- [x] ~~I translated all the newly inserted strings~~

### Description
When re-evaluating quiz questions the method parseTextHintExplanation throws an TypeError: Cannot read property 'trim' of undefined
![image](https://user-images.githubusercontent.com/33764166/61884851-38896c80-aefd-11e9-8277-5fc8c973f00f.png)
This is caused by a wrong RegEx for splitting up the markdown text.

### Steps for Testing
1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Go to a MC quiz that offers the re-evaluate button and click on it
4. Change option contents and save changes
5. Check whether the changes have been applied correctly and the browser console doesn't show the error message 

![image](https://user-images.githubusercontent.com/33764166/61896223-ccb2fe00-af14-11e9-80ec-55a5c00228b2.png)

### Screenshots
There are no UI changes

krusche commented:
Have you checked that the parsing still works in all other cases, e.g. while creating a quiz?

lovaiible commented:
@krusche Yes, parsing in general works for quiz exercises and this parsing method `parseTextHintExplanation` is only used when editing short answer questions or re-evaluating multiple-choice questions, which is fixed with this PR.

lovaiible commented:
@thilo-behnke thank you for your review and remarks! I changed the code based on your suggestion and also added some tests. Can you have a look at it?
